File: ros2_packages/hello_pub_py3/hello_pub_py3/hello_pub.py
import rclpy
from rclpy.node import Node

from std_msgs.msg import String

class HelloPub(Node):
    def __init__(self):
        super().__init__('hello_pub')
        topicName = '/hello'
        self.pub = self.create_publisher(String, topicName, 10)
        self.hello_msg = String()

        # timer
        timer_rate = 10/1000
        self.rate = self.create_timer(timer_rate, self.timerCB)

    def timerCB(self):
        self.hello_msg.data = 'Hello World!'
        self.pub.publish(self.hello_msg)
        self.get_logger().info(self.hello_msg)

def main(args=None):
    rclpy.init(args=args)
    hello_pub = HelloPub()
    try:
        rclpy.spin(hello_pub)
    except KeyboardInterrupt:
        print('killing process...')
    hello_pub.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: ros2_packages/hello_pub_cpp/src/hello_pub.cpp
#include <chrono>
#include <memory>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

using namespace std::chrono_literals;

class HelloPub : public rclcpp::Node
{
public:
	HelloPub();
private:
	std::shared_ptr<std_msgs::msg::String> msg;
	void timerCB();
	rclcpp::TimerBase::SharedPtr timer;
	rclcpp::Publisher<std_msgs::msg::String>::SharedPtr pub;
};
HelloPub::HelloPub() : Node("hello_pub")
{
	msg = std::make_shared<std_msgs::msg::String>();
	pub = this->create_publisher<std_msgs::msg::String>("/hello", 10);
	timer = this->create_wall_timer(100ms, std::bind(&HelloPub::timerCB, this));
}

void HelloPub::timerCB()
{
	msg->data = "Hello World!";
	RCLCPP_INFO(this->get_logger(), "Publishing: %s", msg.data.c_str());
	pub->publish(*msg);
}

int main(int argc, char *argv[])
{
	rclcpp::init(argc, argv);
	rclcpp::spin(std::make_shared<HelloPub>());
	rclcpp::shutdown();
	
	return 0;
}
